pkg/api/service/annotations.go

```diff
@@ -25,4 +25,9 @@ const (
 	//
 	// Not all cloud providers support this annotation, though AWS & GCE do.
 	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
+
+	// An annotation that denotes if this Service desires to only route to local
+	// endpoints. Typically used to preserve source IP in loadbalanced Services.
+	AnnotationOnlyNodeLocalEndpoints = "service.alpha.kubernetes.io/cloud-lb-only-node-local-endpoints"
+	AnnotationHealthCheckNodePort    = "service.alpha.kubernetes.io/healthcheck-nodeport-node-local-endpoints"
```

Comment:
You are thinking we can re-use this functionality - the second annotation by itself would turn on the listener and enable them to poll for the endpoints ? The code currently expects the first annotation to be set and only then uses the second one.
I can file a PR for this feature.